Flow.Launcher/QuickSwitch/NativeHelper.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace Flow.Launcher.QuickSwitch
+{
+    public static class NativeHelper
+    {
+
+
+        public const uint WINEVENT_OUTOFCONTEXT = 0;
+        public const uint EVENT_SYSTEM_FOREGROUND = 3;
+
+        [DllImport("user32.dll")]
+        public static extern unsafe IntPtr SetWinEventHook(uint eventMin, uint eventMax, IntPtr hmodWinEventProc,
+            delegate* unmanaged<IntPtr, uint, IntPtr, int, int, uint, uint, void> lpfnWinEventProc,
+            uint idProcess, uint idThread, uint dwFlags);
+
+        [DllImport("user32")]
+        public static extern IntPtr GetForegroundWindow();
```

Comment:
Quick design questions: 
- Would it ever make sense for a plugin to also query which *other* explorer windows are open? Stuff like having two explorer windows open side by side...?
- And, *how* should the case be handled where the user clearly has one explorer window open, but it's not the topmost window? Like having the explorer window opened, but being focused on the task manager?

Reply:
> Would it ever make sense for a plugin to also query which other explorer windows are open? Stuff like having two explorer windows open side by side...?

Then only sync the path that is moved foreground. I didn't design this feature, but just try to mimic the design.

Here's the original design. https://www.listary.com/quick-switch

> And, how should the case be handled where the user clearly has one explorer window open, but it's not the topmost window? Like having the explorer window opened, but being focused on the task manager?

It may be useful to sync explorer path that is not at the foreground. But I guess the case you describe is somewhat different. The original design gives user control about what they want to sync or when they want to sync the explorer path. Therefore, they need to put the explorer up to front to notify which path they want to sync for.

Reply:
Oh, I understand! Then all of this makes perfect sense.